vapi/rest/client.go

```diff
@@ -63,6 +63,36 @@ func NewClient(c *vim25.Client) *Client {
 	return &Client{sc, ""}
 }
 
+type marshaledClient struct {
+	SoapClient *soap.Client
+	SessionID  string
+}
+
+func (c *Client) MarshalJSON() ([]byte, error) {
```

Comment:
Right, they'll be the same if Terraform moves to using `session.Cache` instead of its own.
Also note that govmomi has switched to using header auth for REST, as cookie auth for REST will be deprecated in soon in vSphere.  See #1884

Reply:
Interesting, I wasn't aware of the deprecation, thanks for letting me know.

Well at least the two comments raised are on the vsphere-provider's development radar and on record. If it does lag behind the latest release of GOVC, then I'll try and make sure its added to the [providers docs.](https://github.com/terraform-providers/terraform-provider-vsphere/blob/ad0ce41fc7ab8eb7926e881a19a5d4056aaa8fca/website/docs/index.html.markdown#govcterraform-session-interoperability)

Reply:
I'd be ok switching back to sha1 so govc and TF can continue to share soap sessions.  Then after TF switches to using cache.Session (assuming it does), we can go back to 256 to make the linters happy.

Reply:
Sounds like a reasonable compromise and a steady integration path 👍